src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -384,6 +387,10 @@ private function getDefinitionSchema(string $resourceClass, ResourceMetadata $re
             $normalizedPropertyName = $this->nameConverter ? $this->nameConverter->normalize($propertyName) : $propertyName;
 
             if ($propertyMetadata->isRequired()) {
+                if (false === $propertyMetadata->isWritable()) {
+                    throw new RuntimeException(sprintf('The property "%s" of the resource "%s" can not be required and read-only at the same time.', $propertyName, $resourceClass));
```

Comment:
This is really problematic, because read-only properties can be "required" too, i.e. they cannot be set to null. Swagger operates in a different context than ours.

For example, we set `withRequired(true)` here: https://github.com/api-platform/core/blob/0f886474d47bad63cca6844557a919e691a356fd/src/Bridge/Symfony/Validator/Metadata/Property/ValidatorPropertyMetadataFactory.php#L61